src/sync/schema/mod.rs

```diff
@@ -149,6 +149,21 @@ impl SyncSchema {
             })
             .unwrap_or_default()
     }
+
+    // Check whether this and the other sync schemas are the same.
+    //
+    // This is a narrower check than the equality check, since it checks for the sync column roles,
+    // names, data types and order (and doesn't inspect the underlying arrows field equality).
+    pub fn same(&self, other: &SyncSchema) -> bool {
```

Comment:
```suggestion
    pub fn is_compatible_with(&self, other: &SyncSchema) -> bool {
```

or `has_same_sync_schema`, `matches_sync_columns`